### ISSUE TITLE
Conditioning InterCTC: self.conditioning_layer is None!

### ISSUE THREAD

**Issue by david-gimeno** [espnet/espnet#5230]
According to [(Nozaki and Kumatsu, 2021)](https://www.isca-speech.org/archive/pdfs/interspeech_2021/nozaki21_interspeech.pdf), remarkable results were found when relaxing the conditional independence assumption of CTC via the incorporation of InterCTC blocks in certain layers of the encoder. These InterCTC blocks is defined as following:

                                                                          inter_ctc_loss
                                                                                   ^
                                                                                   |
          Linear(encoder_size, vocab_size) --> Softmax() --> Linear(vocab_size, encoder_size)*

I inspect the code of the conformer and the ESPNet's implementations seems to follow this paper but there is one thing I do not understand: if we look [this line of code](https://github.com/espnet/espnet/blob/master/espnet2/asr/encoder/conformer_encoder.py#L295) , the `self.conditioning_layer` is always set to None, even if the boolean variable that specifies whether using this conditioning approach is set to True.  Can you please explaining me where this layer is defined? In my opinion it should be `= nn.Linear(vocab_size, output_size)` .

Thanks in advance!

* In order to integrate this tensor via a residual connection into the encoder, i.e., conditioning the encoder.
                                                          
https://github.com/espnet/espnet/blob/master/espnet2/asr/encoder/conformer_encoder.py#L295

**Comment by sw005320** [comment]
Thanks for the report, @david-gimeno!

@wanchichen, can you answer it for me, since you touched this part recently?

**Comment by YosukeHiguchi** [comment]
Hi @david-gimeno, I can answer this question.
The linear layer for self-conditioned CTC is defined [here](https://github.com/espnet/espnet/blob/master/espnet2/asr/espnet_model.py#L108).

**Comment by david-gimeno** [comment]
As always thank you very much @YosukeHiguchi ! I was diving into the code all day and I did not understand it.

Have a nice day!
